patches/tModLoader/Terraria/ModLoader/Default/UnloadedTilesSystem.cs

```diff
@@ -7,111 +7,180 @@ namespace Terraria.ModLoader.Default
+				["tileList"] = tileInfos.Select(info => info?.Save() ?? new TagCompound()).ToList(),
+				["tilePosIndex"] = tileInfoMap.Select(pair => new TagCompound {
+					["posID"] = pair.Key,
+					["infoID"] = pair.Value
+				}).ToList(),
+				["wallList"] = wallInfos.Select(info => info?.Save() ?? new TagCompound()).ToList(),
+				["wallPosIndex"] = wallInfoMap.Select(pair => new TagCompound {
+					["posID"] = pair.Key,
+					["infoID"] = pair.Value
+				}).ToList(),
+				["chestList"] = chestInfos.Select(info => info?.Save() ?? new TagCompound()).ToList(),
+				["chestPosIndex"] = chestInfoMap.Select(pair => new TagCompound {
+					["posID"] = pair.Key,
+					["infoID"] = pair.Value
+				}).ToList(),
 			};
 		}
 
 		public override void LoadWorldData(TagCompound tag) {
-			List<ushort> canRestore = new List<ushort>();
-			bool canRestoreFlag = false;
-
-			foreach (var infoTag in tag.GetList<TagCompound>("list")) {
-				if (!infoTag.ContainsKey("mod")) {
-					infos.Add(null);
-					canRestore.Add(0);
-					continue;
-				}
+			bool[] canRestoreFlag = new bool[] { false, false, false };
 
-				string modName = infoTag.GetString("mod");
-				string name = infoTag.GetString("name");
-				bool frameImportant = infoTag.ContainsKey("frameX");
-				var info = frameImportant ?
-					new UnloadedTileInfo(modName, name, infoTag.GetShort("frameX"), infoTag.GetShort("frameY")) :
-					new UnloadedTileInfo(modName, name);
+			// Process tiles
+			UpdateUnloaded tileUpdater = new UpdateUnloaded(TilesIndex);
+			tileUpdater.updateInfos(tag.GetList<TagCompound>("tileList"));
+			canRestoreFlag[TilesIndex] = tileUpdater.CanRestoreFlag;
+			tileUpdater.updateMaps(tag.GetList<TagCompound>("tilePosIndex"));
 
-				infos.Add(info);
+			// Process Walls
+			UpdateUnloaded wallUpdater = new UpdateUnloaded(WallsIndex);
+			wallUpdater.updateInfos(tag.GetList<TagCompound>("wallList"));
+			canRestoreFlag[WallsIndex] = wallUpdater.CanRestoreFlag;
+			wallUpdater.updateMaps(tag.GetList<TagCompound>("wallPosIndex"));
 
-				int type = ModContent.TryFind(modName, name, out ModTile tile) ? tile.Type : 0;
+			// Process chests
+			UpdateUnloaded chestUpdater = new UpdateUnloaded(ChestIndex);
+			chestUpdater.updateInfos(tag.GetList<TagCompound>("chestList"));
+			canRestoreFlag[ChestIndex] = chestUpdater.CanRestoreFlag;
+			chestUpdater.updateMaps(tag.GetList<TagCompound>("chestPosIndex"));
 
-				canRestore.Add((ushort)type);
-
-				if (type != 0)
-					canRestoreFlag = true;
-			}
+			// If restoration should occur during this load cycle, then do so
+			RestoreTilesAndWalls(tileUpdater.CanRestore, wallUpdater.CanRestore, chestUpdater.CanRestore, canRestoreFlag);
 
-			if (canRestoreFlag) {
-				RestoreTiles(canRestore);
+			// Cleanup data remnants
+			tileUpdater.cleanupMaps();
+			tileUpdater.cleanupInfos();
+			wallUpdater.cleanupMaps();
+			wallUpdater.cleanupInfos();
+			chestUpdater.cleanupMaps();
+			chestUpdater.cleanupInfos();
```

Comment:
All methods must be PascalCase, read the style guide - https://github.com/tModLoader/tModLoader/wiki/tModLoader-Style-Guide

Reply:
Noted - it appears I switched along the way for that class.
Fixed the casing.